guidance/_parser.py

```diff
@@ -3,8 +3,24 @@
 from ordered_set import OrderedSet
 from ._grammar import Join, Select, Terminal, Null, Byte, ByteRange
 
+
+class ParserException(Exception):
```

Comment:
The introduction of this exception is the key change. For the rest, I appear to have accidentally run `black` on the file. I can unpick this if desired.

Reply:
I like the exception, but can we undo the linting for the time being just to keep git blame clean in this part of the code?

Reply:
Done